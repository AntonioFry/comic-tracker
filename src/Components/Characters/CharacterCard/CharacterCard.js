import React from 'react';
import './CharacterCard.css';
import { Link } from 'react-router-dom';

export const CharacterCard = ({ thumbnail, name, id }) => {
  return (
    <Link className='character-link' to={`/characters/${id}`} >
      {thumbnail === undefined ? null : <div className="thumbnail-container">
        <img className="thumbnail-image" src={`${thumbnail.path}.${thumbnail.extension}`} alt="character preview thumbnail" />
        <h3 className="thumbnail-name">{name}</h3>
      </div>}
    </Link>
  )
}