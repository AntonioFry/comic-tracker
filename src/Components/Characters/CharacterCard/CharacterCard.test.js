import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import { CharacterCard } from './CharacterCard';
import { BrowserRouter } from 'react-router-dom'

describe('CharacterCard', () => {
  let props = {
    thumbnail: 'Spider-Man.JPG',
    id: '1100616',
    name: 'Spider-Man'
  };

  it('Should match the snapshot', () => {
    const tree = renderer.create( 
      <BrowserRouter>
        <CharacterCard {...props} />
      </BrowserRouter>  
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});