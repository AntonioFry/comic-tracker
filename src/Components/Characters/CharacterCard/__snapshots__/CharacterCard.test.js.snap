exports[`CharacterCard Should match the snapshot 1`] = `
<a
  className="character-link"
  href="/characters/1100616"
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }>
  <div
    className="thumbnail-container">
    <img
      alt="character preview thumbnail"
      className="thumbnail-image"
      src="undefined.undefined" />
    <h3
      className="thumbnail-title">
      Spider-Man
    </h3>
  </div>
</a>
`;
